components/scream/src/interface/scream_scorpio_interface.F90

```diff
@@ -0,0 +1,977 @@
+    call register_variable("example_pio_structured_v2.nc","x","answer to space and time",1,(/ "x" /), PIO_double,"x-real")
+    call register_variable("example_pio_structured_v2.nc","y","answer to space and time",1,(/ "y" /), PIO_double,"y-real")
+    call register_variable("example_pio_structured_v2.nc","bar","answer to space and time",3,(/ "x", "y", "time" /), PIO_double,"xyt-real")
+    call register_variable("example_pio_structured_v2.nc","foo","answer to space and time",3,(/ "x", "y", "time" /), PIO_double,"xyt-real")
+    call register_variable("example_pio_structured_v2.nc","bar_flip","answer to space and time",3,(/ "y", "x", "time" /), PIO_double,"yxt-real")
+    call register_variable("example_pio_structured_v2.nc","foo_flip","answer to space and time",3,(/ "y", "x", "time" /), PIO_double,"yxt-real")
+
+    call get_pio_atm_File(trim("example_pio_structured_v2.nc"),current_atm_file,found)
+    ! It is essential that PIO_enddef is called when all dimensions and
+    ! variables have been registered with a specific file.
+    ierr = PIO_enddef(current_atm_file%pioFileDesc)
+    call errorHandle("PIO ERROR: issue arose with PIO_enddef for file"//trim(current_atm_file%filename),ierr)
+
+  end subroutine eam_init_pio_2
+!=====================================================================!
+  ! Register a dimension with a specific pio output file 
+  subroutine register_dimension(pio_atm_filename,shortname,longname,length)
+    character(len=*), intent(in)        :: pio_atm_filename   ! Name of file to register the dimension on.
+    character(len=*), intent(in)        :: shortname,longname ! Short- and long- names for this dimension, short: brief identifier and name for netCDF output, long: longer descriptor sentence to be included as meta-data in file.
+    integer, intent(in)                 :: length             ! Length of the dimension, 0: unlimited (like time), >0 actual length of dimension
+
+    type(pio_atm_output), pointer       :: pio_atm_file
+    type(hist_coord_t), pointer         :: hist_coord
+    type(hist_coord_list), pointer      :: curr=>null(), prev=>null()
+    integer                             :: ierr
+    character(len=150)                  :: errstr
+    logical                             :: found
+
+    ! Make sure the dimension length is reasonable
+    if (length<0) call errorHandle("PIO Error: dimension "//trim(shortname)//", can't have a negative dimension length",-999)
+ 
+    ! Find the pointer for this file
+    call get_pio_atm_File(trim(pio_atm_filename),pio_atm_file,found)
+    if (.not.found) call errorHandle("PIO Error: Could not register dimension "//trim(shortname)//", pio file "//trim(pio_atm_filename)//" not found",-999)
+    ! Get a new dimension pointer in coord_list
+    curr => pio_atm_file%coord_list_top
+    do while (associated(curr))
+      prev => curr
+      curr => prev%next
+    end do
```

Comment:
again, what happens if a dimension with the same name/size is already registered?

Reply:
Good point. I believe PIO issues it's own error on the backend, but probably best to have a check here anyway since it is essentially free.